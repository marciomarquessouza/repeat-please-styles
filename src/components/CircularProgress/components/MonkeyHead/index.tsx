import React, { Component } from 'react';
import { Animated, Image, ImageSourcePropType, Easing } from 'react-native';
import {
	animationFactory,
	IAnimationFactory,
} from '../../../../animation/animationFactory';

interface IMonkeyHeadProps {
	progress: number;
	chartSize: number;
}

export class MonkeyHead extends Component<IMonkeyHeadProps, {}> {
	private bounceAnimation = new Animated.Value(0);

	componentDidMount() {
		this.monkeyHeadAnimation();
	}

	monkeyHeadAnimation = () => {
		const monkeyHeadBounce: IAnimationFactory[] = [
			{
				animationObject: this.bounceAnimation,
				toValue: 1,
				duration: 1000,
				easing: Easing.bounce,
			},
		];
		animationFactory(monkeyHeadBounce);
	};

	monkeyImage = (progress: number): ImageSourcePropType => {
		const monkeyImages = [
			require('../../../../../assets/images/SeriousMonkey.png'),
			require('../../../../../assets/images/MouthMonkey.png'),
			require('../../../../../assets/images/SmileMonkey.png'),
		];
		if (progress >= 0.75) {
			return monkeyImages[2];
		} else if (progress >= 0.25 && progress < 0.75) {
			return monkeyImages[1];
		} else {
			return monkeyImages[0];
		}
	};

	render() {
		const SIZE = 50;
		const { PI, sin, cos } = Math;
		const { chartSize, progress } = this.props;
		const radius = SIZE / 2;
		const chartRadius = chartSize / 2;
		const posX = -cos(progress * 2 * PI) * (chartRadius + radius + 10);
		const posY = -sin(progress * 2 * PI) * (chartRadius + radius + 10);

		return (
			<Animated.View
				style={{
					position: 'absolute',
					justifyContent: 'center',
					alignItems: 'center',
					alignSelf: 'center',
					opacity: this.bounceAnimation,
					transform: [
						{ translateX: posX },
						{ translateY: posY },
						{ scale: this.bounceAnimation },
					],
				}}>
				<Image
					source={this.monkeyImage(progress)}
					style={{
						width: SIZE,
						height: SIZE,
					}}
				/>
			</Animated.View>
		);
	}
}
