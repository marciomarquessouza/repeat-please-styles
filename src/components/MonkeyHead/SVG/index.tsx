export * from './MouthSVG';
export * from './EyesBrownSVG';
export * from './EyesSVG';
export * from './FaceSVG';
export * from './HairSVG';
export * from './NooseSVG';
export * from './EarsSVG';
