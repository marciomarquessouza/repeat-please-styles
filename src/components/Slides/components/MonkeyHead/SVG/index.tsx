import React, { useState, useEffect } from 'react';
import Svg from 'react-native-svg';
import { Mouth } from './MouthSVG';
import { EyeBrow } from './EyesBrownSVG';
import { Eyes } from './EyesSVG';
import { Face } from './FaceSVG';
import { Hair } from './HairSVG';
import { Noose } from './NooseSVG';
import { Ears } from './EarsSVG';

export enum Mood {
	sad = 'sad',
	curious = 'curious',
	happy = 'happy',
}

interface IMonkeyHeadProps {
	width: number;
	height: number;
	index: number;
}

export const MonkeyHeadSVG = (props: IMonkeyHeadProps): JSX.Element | null => {
	const [moodID, setMoodID] = useState(props.index);
	useEffect(() => setMoodID(props.index), [props.index]);
	let mood: Mood;
	switch (moodID) {
		case 0:
			mood = Mood.sad;
			break;
		case 1:
			mood = Mood.curious;
			break;
		case 2:
			mood = Mood.happy;
			break;
		default:
			return null;
	}

	return (
		<Svg fill="none" {...props}>
			{Face}
			{Eyes[mood]}
			{Hair}
			{Noose}
			{Ears}
			{Mouth[mood]}
			{EyeBrow[mood]}
		</Svg>
	);
};
