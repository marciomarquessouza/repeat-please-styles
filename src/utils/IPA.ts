export const IPA = {
	ʌ: {
		symbol: 'ʌ',
		type: 'vowel',
		examples: ['cUp', 'lUck'],
	},
	'ɑ:': {
		symbol: 'ɑ:',
		type: 'vowel',
		examples: ['Arm', 'fAther'],
	},
	æ: {
		symbol: 'æ',
		type: 'vowel',
		examples: ['cAt', 'blAck'],
	},
	e: {
		symbol: 'e',
		type: 'vowel',
		examples: ['mEt', 'bEd'],
	},
	ə: {
		symbol: 'ə',
		type: 'vowel',
		examples: ['Away', 'cinemA'],
	},
	'ɜ:': {
		symbol: 'ɜ:',
		type: 'vowel',
		examples: ['tURn', 'lEARn'],
	},
	ɪ: {
		symbol: 'ɪ',
		type: 'vowel',
		examples: ['hIt', 'sITTING'],
	},
	'i:': {
		symbol: 'i:',
		type: 'vowel',
		examples: ['SEE', 'HEAT'],
	},
	ɒ: {
		symbol: 'ɒ',
		type: 'vowel',
		examples: ['HOT', 'ROCK'],
	},
	'ɔ:': {
		symbol: 'ɔ:',
		type: 'vowel',
		examples: ['CALL', 'FOUR'],
	},
	ʊ: {
		symbol: 'ʊ',
		type: 'vowel',
		examples: ['PUT', 'COULD'],
	},
	'u:': {
		symbol: 'u:',
		type: 'vowel',
		examples: ['BLUE', 'FOOD'],
	},
	aɪ: {
		symbol: 'aɪ',
		type: 'vowel',
		examples: ['FIVE', 'EYE'],
	},
	aʊ: {
		symbol: 'aʊ',
		type: 'vowel',
		examples: ['NOW', 'OUT'],
	},
	eɪ: {
		symbol: 'eɪ',
		type: 'vowel',
		examples: ['SAY', 'EIGHT'],
	},
	oʊ: {
		symbol: 'oʊ',
		type: 'vowel',
		examples: ['GO', 'HOME'],
	},
	ɔɪ: {
		symbol: 'ɔɪ',
		type: 'vowel',
		examples: ['BOY', 'JOIN'],
	},
	eə: {
		symbol: 'eə',
		type: 'vowel',
		examples: ['WHERE', 'AIR'],
	},
	ɪə: {
		symbol: 'ɪə',
		type: 'vowel',
		examples: ['NEAR', 'HERE'],
	},
	ʊə: {
		symbol: 'ʊə',
		type: 'vowel',
		examples: ['PURE', 'TOURIST'],
	},
};
