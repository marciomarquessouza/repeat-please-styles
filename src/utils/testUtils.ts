import { ShallowWrapper } from 'enzyme';

export const findByDataTest = (
	wrapper: ShallowWrapper,
	dataTest: string,
): ShallowWrapper => {
	return wrapper.find(`[data-test="${dataTest}"]`);
};
