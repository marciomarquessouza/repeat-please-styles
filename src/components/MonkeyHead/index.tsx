import React from 'react';
import Svg from 'react-native-svg';
import { Ears, Mouth, EyeBrow, Eyes, Face, Hair, Noose } from './SVG';
import { View, ViewStyle } from 'react-native';

export enum Mood {
	sad = 'sad',
	curious = 'curious',
	happy = 'happy',
}

export interface IMonkeyHeadProps {
	width?: number;
	height?: number;
	mood: Mood;
	style?: ViewStyle;
}

export const MonkeyHead = ({
	mood,
	style,
	width,
	height,
}: IMonkeyHeadProps): JSX.Element => (
	<View {...{ style }}>
		<Svg
			width={width || 260}
			height={height || 290}
			viewBox="0 0 260 290"
			fill="none">
			{Face}
			{Eyes[mood]}
			{Hair}
			{Noose}
			{Ears}
			{Mouth[mood]}
			{EyeBrow[mood]}
		</Svg>
	</View>
);
