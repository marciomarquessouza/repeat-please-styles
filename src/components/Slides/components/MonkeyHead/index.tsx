import React, { useState, useEffect } from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';
import { MonkeyHeadSVG } from './SVG';

interface IMonkeyHeadProps {
	index: number;
	height: number;
	onPress?: () => void;
}

export const MonkeyHead = (props: IMonkeyHeadProps): JSX.Element => {
	const [index, setIndex] = useState(props.index);

	useEffect(() => setIndex(props.index), [props.index]);

	const { height, onPress } = props;
	const monkeyHeadSize = { width: 252, height: 280 };
	const top = height / 2 - monkeyHeadSize.height / 2 - 30;

	return (
		<View style={[styles.monkeyHeadDesign, { top }]}>
			<TouchableOpacity onPress={onPress} activeOpacity={1}>
				<MonkeyHeadSVG
					height={monkeyHeadSize.height}
					width={monkeyHeadSize.width}
					index={index}
				/>
			</TouchableOpacity>
		</View>
	);
};

const styles = StyleSheet.create({
	monkeyHeadDesign: {
		justifyContent: 'center',
		alignItems: 'center',
		alignSelf: 'center',
		position: 'absolute',
	},
});
