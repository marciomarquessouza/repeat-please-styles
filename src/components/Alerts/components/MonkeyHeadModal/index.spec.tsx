import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { MonkeyHeadModal, IMonkeyHeadModal } from './';
import { findByDataTest } from '../../../../utils/testUtils';
import { Mood } from '../../../MonkeyHead';

const setup = (props: IMonkeyHeadModal): ShallowWrapper => {
	return shallow(<MonkeyHeadModal {...props} />);
};

describe('Monkey Head Modal', () => {
	it('should render the component properly', () => {
		const wrapper = setup({ mood: Mood.curious });
		const monkeyHead = findByDataTest(wrapper, 'monkeyHead');
		expect(monkeyHead).toHaveLength(1);
		expect(monkeyHead).toMatchSnapshot();
	});
});
