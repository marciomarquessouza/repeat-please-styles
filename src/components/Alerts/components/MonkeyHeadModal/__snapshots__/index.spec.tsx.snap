// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monkey Head Modal should render the component properly 1`] = `ShallowWrapper {}`;
